src/driver.toit

```diff
@@ -144,34 +160,50 @@ class Driver:
     nbtcfg |= PS2 - 1
     nbtcfg <<= 8
     nbtcfg |= PS2 - 1
-    write_u32_ NBTCFG_REG_ nbtcfg
+    write_u32_ device NBTCFG_REG_ nbtcfg
 
-    enter_mode_ MODE_NORMAL_CAN_
+    enter_mode_ device MODE_NORMAL_CAN_
 
   /**
   Configures and enables a filter.
 
   Index is 0-based.
   */
   add_filter index/int mask/int obj/int:
-    write_u32_ (filter_mask_reg_ index) mask
-    write_u32_ (filter_obj_reg_ index) obj
+    device_mutex_.do:
+      add_filter_ device_ index mask obj
+
+  add_filter_ device/spi.Device index/int mask/int obj/int:
+    write_u32_ device (filter_mask_reg_ index) mask
+    write_u32_ device (filter_obj_reg_ index) obj
     flt_con0 := 1 << 7 // Filter is enabled
     // Add messages to receive FIFO.
     flt_con0 |= RECEIVE_FIFO_INDEX_
-    write_u8_ (filter_con_reg_ index) 0 flt_con0
+    write_u8_ device (filter_con_reg_ index) 0 flt_con0
 
   /**
   Run the interrupt rutine. Without this, the driver will not be able to
     send or receive messages.
 
   This methods blocks for the lifecycle of the driver.
+
```

Comment:
rutine -> routine
methods -> method
of the driver. -> of the driver, and should be called on a dedicated thread.